### PR TITLE
update(Proofreader): Handle delete replacement

### PR DESCRIPTION
**Pull request by al-chen**
to: @milesj @stefhatcher @hayes @alecklandgraf
cc: @QecTu @tyn2

## Description

Handle the case in Proofreader when error replacement is `''` by representing it with `(Delete)`. 

## Motivation and Context

When error replacement is an empty string, it ostensibly means we should delete the error. We should make that more obvious rather than display a blank row. 

## Testing

Updated unit tests. Verified locally in storybook. 

## Screenshots
Before:
Screen Shot 2019-06-06 at 6 04 11 PM

After:
Jun-07-2019 10-51-51

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by tyn2** [comment]
@al-chen Looks good, can we just update the text to `Delete this note`?

File: packages/core/src/components/TextArea/Proofreader/ErrorMenu.tsx [hunk 1]
```diff
@@ -16,6 +16,17 @@ export default class ErrorMenu extends React.Component<Props> {
     this.props.onReplaceText(this.props.error, replacement);
   };
 
+  private replacementPhrase = (replacement: string) => {
```
Comment (stefhatcher) [inline packages/core/src/components/TextArea/Proofreader/ErrorMenu.tsx]: is this `private` needed? usually just added for handlers

Comment (al-chen) [reply packages/core/src/components/TextArea/Proofreader/ErrorMenu.tsx]: It doesn't seem necessary, but to me (admittedly neither a TS nor Lunar expert), I'm not sure there's harm in adding it to encapsulate implementation within this component. Can remove if convention and style disagree.

**Comment by tyn2** [comment]
@al-chen Can we actually do `Delete this note` without the parentheses?

File: packages/core/src/components/TextArea/Proofreader/ErrorMenu.tsx (hunk 1 above)
Comment (stefhatcher) [reply packages/core/src/components/TextArea/Proofreader/ErrorMenu.tsx]: not a blocker! did a rough scan of lunar and it is mostly in front of handlers, but some other methods too. seems fine!

**Comment by milesj** [comment]
Why don't we just filter out errors that are `''`?

**Comment by al-chen** [comment]
@milesj this change isn't for errors that are `''`, it's for replacements to the errors that are `''`. Seems intuitive to me that if you want to replace text with `''`, you _probably_ want to delete the original text - we're making that more obvious here.

**Comment by milesj** [comment]
@al-chen The after image helps. Curious where the "No need to note this action..." message come from? Not sure "note" makes enough sense in this context.

**Comment by tyn2** [comment]
@milesj @al-chen The `No need to note this action…` tooltip is from our content strategist and I'm assuming "note" is taken derived from "admin notes". Or do you mean the `Delete this note` text is ambiguous?

**Comment by al-chen** [comment]
@milesj @stefhatcher hello! if this LG, can we merge?

**Pull request merged**